nomos-services/cryptarchia-consensus/src/leadership.rs

```diff
@@ -126,6 +135,10 @@ impl Leader {
 
         None
     }
+
+    pub(crate) fn notes(&self, header_id: &HeaderId) -> Option<&Vec<NoteWitness>> {
```

Comment:
Would it make sense to rater return an `impl Iterator` here? Then the consumer can collect it however they want to.

Reply:
`self.notes` is already a `Vec`, do you think it's actually worth casting it to return it here?

Reply:
Probaly `Option<&[NoteWitness]>` is fine. From that can be transformer into iter if needed, or be indexed (which probably is not needed but anyway...).